commcare_connect/form_receiver/processor.py

```
@@ -364,30 +365,34 @@ def get_or_create_deliver_unit(app, unit_data):
-def get_opportunity(*, learn_app=None, deliver_app=None):
-    if not learn_app and not deliver_app:
-        raise ValueError("One of learn_app or deliver_app must be provided")
-
-    kwargs = {}
-    if learn_app:
-        kwargs = {"learn_app": learn_app}
-    if deliver_app:
-        kwargs = {"deliver_app": deliver_app}
+def get_opportunity(domain, hq_server, deliver_app_id=None, learn_app_id=None):
+    if not (learn_app_id or deliver_app_id):
+        raise ValueError("One of learn_app_id or deliver_app_id along with domain must be provided")
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)